iot-hub/Samples/device/DeviceReconnectionSample/DeviceReconnectionSample.cs

```diff
@@ -106,6 +119,21 @@ private async Task InitializeAndOpenClientAsync()
                     // OpenAsync() is an idempotent call, it has the same effect if called once or multiple times on the same client.
                     await s_deviceClient.OpenAsync();
                     _logger.LogDebug($"Opened the client instance.");
+
+                    switch (_transportType)
+                    {
+                        case TransportType.Amqp:
+                        case TransportType.Amqp_Tcp_Only:
+                        case TransportType.Amqp_WebSocket_Only:
+                        case TransportType.Mqtt:
+                        case TransportType.Mqtt_Tcp_Only:
+                        case TransportType.Mqtt_WebSocket_Only:

```

Comment:
this could also have been if (_transportType is not Http1) -> subscribe to c2d; but I wanted to call out that all these transport protocols supports c2d receive callback; so I went with the switch case

Reply:
Adding a case for Http1 and returning "OperationNotAlllowed" exception would make this switch case better.